content/guides/concepts/cid.md

```diff
@@ -5,18 +5,33 @@ menu:
         parent: concepts
 ---
 
-A *content identifier* is a value that addresses a single piece of content in IPFS. It is mainly a cryptographic hash of the content, but is encoded as a [multihash](https://github.com/multiformats/multihash) and [multicodec](https://github.com/multiformats/multicodec). (Note: older CIDs have a different design — see [version 0](#version-0) below.)
+A *content identifier*, or CID, is a label used as a standardized way of pointing to material in IPFS. It doesn't indicate _where_ the content is stored, but it forms a kind of address out of the content itself.  CIDs are consistently compact, regardless of the size of underlying content.
 
-<!-- TODO: explain more of the details of how CID v1 is composed here. -->
+CIDs are based on the content's [cryptographic hash](concepts/hashes).  As a result, any difference in content will produce a different CID.  Any IPFS node having the content will be able to match the hash and be able to retrieve the original content.  
```

Comment:
Fact-check: will that node be able to match the hash if it has the content?

Reply:
Mostly. The caveat here is that the node has to have the content stored with the same hash algorithm (NOT the same CID — work is currently happening to make sure different *base-encodings* of the same hash will point to the same content on disk in ipfs/go-ipfs#5231 — and the codec part of a CID is not involved in lookup, it’s only a hint for parsing the content once it’s found and received).

For example, say you have the content `"Hello"`. If you hash it with SHA-256, you can make differently encoded CIDs of that:

- Base 58 (default): [`zb2rhdYtXM8X3Jfsm6VrmXnmcSqtfgHZbhYRJ32ENkmARL78K`](http://cid-utils.ipfs.team/#zb2rhdYtXM8X3Jfsm6VrmXnmcSqtfgHZbhYRJ32ENkmARL78K)
- Base 32: [`bafkreidgubc3iuqqfrm5qqhmbf6vtwkgpyj2h42pmskokop72mwbxm27da`](http://cid-utils.ipfs.team/#bafkreidgubc3iuqqfrm5qqhmbf6vtwkgpyj2h42pmskokop72mwbxm27da)

In current versions of IPFS, they will be treated as totally separate content. In the next version (see issue linked above), they’ll match the *same* content. However, if you used a different hash algorithm:

- SHA-256 Base 58: [`zb2rhdYtXM8X3Jfsm6VrmXnmcSqtfgHZbhYRJ32ENkmARL78K`](http://cid-utils.ipfs.team/#zb2rhdYtXM8X3Jfsm6VrmXnmcSqtfgHZbhYRJ32ENkmARL78K) (same as above)
- SHA-512 Base 58: [`zB7NCgi6ywUVNus2k55hDjeyHPmJrtY3eMB8S9T2Unjyp9Lc3H9yvTq6b4nXieFCcg4awXzKRrSo5GULNR1TMtSbNhGKp`](http://cid-utils.ipfs.team/#zB7NCgi6ywUVNus2k55hDjeyHPmJrtY3eMB8S9T2Unjyp9Lc3H9yvTq6b4nXieFCcg4awXzKRrSo5GULNR1TMtSbNhGKp)

Those will *not* match the same content, because they use a different hashing algorithm.

Maybe instead of the last sentence, say something like:

> Each IPFS node keeps a list of hashes for all the content it stores; when it receives a request for a CID, it extracts the hash from the CID and checks it against the list, then returns the associated content if it’s found.